qualtran/_infra/registers.py

```diff
@@ -181,14 +189,14 @@ def __init__(self, registers: Iterable[Register]):
         self._rights = _dedupe((reg.name, reg) for reg in self._registers if reg.side & Side.RIGHT)
 
     @classmethod
-    def build(cls, **registers: int) -> 'Signature':
+    def build(cls, **registers: QDTypeT) -> 'Signature':
```

Comment:
this should be `QDType`, right? Signature isn't generic

Reply:
Some philosophy:

I'd like to keep `Signature.build` really dead simple where conciseness is key and you don't want to import anything and direct any cases that don't work with `Signature.build` to the structured list-of-registers thing. We've gotten pretty far in the library with the number "1" meaning qubit and any number greater than 1 meaning "unsigned integer with this many bits". I'd like to keep this quaint set of assumptions in `Signature.build`. 

```python
def build(cls, **registers: int):
  if reg == 1: return Register(regname, QBit(), THRU)
  return Register(regname, QInt(bitsize=reg), THRU)
```

Reply:
I've been thinking of QAny as the default here, but, sure this makes sense. We're taking QInt as a signed integer so I was a little hesitant to make it the default.

Reply:
QAny would work too. I'll leave that up to you. There's a bit of fineesse  where the default should be chosen to simplify the cases we think should be simple rather than a super principled argument.